test-utils/state-viewer/src/apply_chain_range.rs

```diff
@@ -151,6 +173,25 @@ pub fn apply_chain_range(
 
                 num_receipt = receipts.len();
                 num_tx = chunk.transactions().len();
+                if only_contracts {
+                    let mut has_contracts = false;
+                    for tx in chunk.transactions() {
+                        for action in &tx.transaction.actions {
+                            has_contracts = has_contracts || match action {
+                                Action::FunctionCall(_) => {
+                                    true
+                                }
+                                Action::DeployContract(_) => {
+                                    true
+                                }
+                                _ => {
+                                    false
+                                }
+                            }
+                        }
+                    }
```

Comment:
+1 to `iter().any()`

Reply:
> LGTM, but do we have any idea of how effective is this? We skip the block if **all** transactions are not function calls, and there are many txes in a block. It might be that we skip very few blocks this way....

Per my analysis it seems we skip significant amount of blocks, could add stats for that.

Reply:
Well, not sure how to write it with `any` where we need to visit both iterators. This version at least look symmetrical, but I'm open to proposals.